pkgs/native_toolchain_c/lib/src/cbuilder/run_cbuilder.dart

```diff
@@ -102,23 +112,37 @@ class RunCBuilder {
       compiler.uri.resolve('../sysroot/');
 
   Future<void> run() async {
-    final compiler_ = await compiler();
-    final compilerTool = compiler_.tool;
-    if (compilerTool == appleClang ||
-        compilerTool == clang ||
-        compilerTool == gcc) {
-      await runClangLike(compiler: compiler_);
-      return;
+    if (linkerOptions != null) {
+      final linker_ = await linker();
+      final linkerTool = linker_.tool;
+      if (linkerTool == appleClang ||
+          linkerTool == clang ||
+          linkerTool == gcc ||
+          linkerTool == gnuLinker) {
+        await runClangLike(compiler: linker_);
```

Comment:
`compiler: linker_` looks odd, maybe rename the param to `tool` so it's `tool: linker_`.